src/TaylorBot.Net/Core/src/TaylorBot.Net.Commands/Parsers/Users/ParsedUser.cs

```diff
@@ -29,10 +29,13 @@ public UserParser(ITaylorBotClient taylorBotClient, IUserTracker userTracker)
             }
 
             var user = await _taylorBotClient.ResolveRequiredUserAsync(new(optionValue.Value.GetString()!));
+            var member = context.Guild == null
+                ? user
+                : await _taylorBotClient.ResolveGuildUserAsync(context.Guild, new(optionValue.Value.GetString()!));
```

Comment:
I think it's OK to change this parser to get member whenever possible. We should avoid doing 2 fetches though, let's change it so we only fetch the user when it's not in a guild or the member doesn't exist